clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2715,6 +2715,20 @@ mlir::LogicalResult cir::ThrowOp::verify() {
   return failure();
 }
 
+//===----------------------------------------------------------------------===//
+// AtomicCmpXchg
+//===----------------------------------------------------------------------===//
+
+LogicalResult cir::AtomicCmpXchg::verify() {
+  auto pointeeType = getPtr().getType().getPointee();
+
+  if (pointeeType != getExpected().getType() or
```

Comment:
```suggestion
  if (pointeeType != getExpected().getType() ||
```
I was surprised not to see a specific recommendation about this in the coding standards, but for the sake of consistency, I think it's best to stick with C-style logical operators.